File: pyhon_src/handle_turtlesim/scripts/sub.py
#! /usr/bin/env python3

import rospy
from turtlesim.msg import Pose

def pose_cb(msg):
    rospy.loginfo("i received: [%s, %s], direction: %s" %(msg.x, msg.y, msg.theta));

def main():
    rospy.init_node("sub_pose")

    rospy.Subscriber("turtle1/pose", Pose, pose_cb)


    rospy.spin()





if __name__ == "__main__":
    main()

File: cpp_src/handle_turtlesim/src/sub.cpp
#include <ros/ros.h>
#include <turtlesim/Pose.h>

void processMsg(const turtlesim::Pose& msg) {
    ROS_INFO("position: [%f, %f], direction: %f", msg.x, msg.y, msg.theta);
}

int main(int argc, char** argv) {
    ros::init(argc, argv, "sub_pose");
    ros::NodeHandle nh;
    ros::Subscriber sub = nh.subscribe("turtle1/pose", 1000, &processMsg);

    ros::spin();
    return 0;
}